zookeeper-server/src/test/java/org/apache/zookeeper/server/quorum/QuorumPeerConfigTest.java

```diff
@@ -164,6 +164,31 @@ public void testJvmPauseMonitorConfigured() throws IOException, ConfigException
         assertTrue(quorumPeerConfig.isJvmPauseMonitorToRun());
     }
 
+    /**
+     * Test case for https://issues.apache.org/jira/browse/ZOOKEEPER-3721
+     */
+    @Test
+    public void testParseBoolean() throws IOException, ConfigException {
+        QuorumPeerConfig quorumPeerConfig = new QuorumPeerConfig();
+        Properties zkProp = getDefaultZKProperties();
+
+        zkProp.setProperty("localSessionsEnabled", "true");
+        quorumPeerConfig.parseProperties(zkProp);
+        assertEquals(true, quorumPeerConfig.areLocalSessionsEnabled());
+
+        zkProp.setProperty("localSessionsEnabled", "false");
+        quorumPeerConfig.parseProperties(zkProp);
+        assertEquals(false, quorumPeerConfig.areLocalSessionsEnabled());
+
+        zkProp.setProperty("localSessionsEnabled", "yes");
+        try {
+            quorumPeerConfig.parseProperties(zkProp);
+            fail("Must throw exception as 'yes' is not accpetable for parseBoolean!");
```

Comment:
we also need to assert `True` or `False` ?

Reply:
Do you mean things like:
```
zkProp.setProperty("localSessionsEnabled", "True");
quorumPeerConfig.parseProperties(zkProp);
assertEquals(true, quorumPeerConfig.areLocalSessionsEnabled());

zkProp.setProperty("localSessionsEnabled", "False");
quorumPeerConfig.parseProperties(zkProp);
assertEquals(false, quorumPeerConfig.areLocalSessionsEnabled());
```
to test whether ignoring case can work?

Reply:
Do you mean things like:
```
zkProp.setProperty("localSessionsEnabled", "True");
quorumPeerConfig.parseProperties(zkProp);
assertEquals(true, quorumPeerConfig.areLocalSessionsEnabled());

zkProp.setProperty("localSessionsEnabled", "False");
quorumPeerConfig.parseProperties(zkProp);
assertEquals(false, quorumPeerConfig.areLocalSessionsEnabled());
```
to test whether ignoring case can work?

Reply:
Yes it is